File: apps/buyer-mobile/lib/features/catalog/data/category_images.dart
/// Bundled top-category illustration assets, keyed by category slug. Mirrors the
/// web `lib/category-images.ts`. The 7 top categories are a fixed taxonomy, so
/// bundling beats a network fetch (instant, works offline, no per-icon round-trip
/// on DRC 2G/3G). Subcategories / future categories without an asset fall back
/// to the emoji.
///
/// To add a category image: drop `assets/categories/<slug>.png`, add one line,
/// and keep the `pubspec.yaml` `assets/categories/` entry.
const Map<String, String> _categoryImages = {
  'supermarche': 'assets/categories/supermarche.png',
  'telephones-et-accessoires': 'assets/categories/telephones-et-accessoires.png',
  'electromenager': 'assets/categories/electromenager.png',
  'mode': 'assets/categories/mode.png',
  'beaute-et-sante': 'assets/categories/beaute-et-sante.png',
  'construction-et-bricolage': 'assets/categories/construction-et-bricolage.png',
  'automobile-et-moto': 'assets/categories/automobile-et-moto.png',
};

String? categoryImageAsset(String? slug) {
  if (slug == null) return null;
  return _categoryImages[slug];
}

File: apps/buyer-web/src/lib/category-images.ts
/**
 * Top-category illustration images (bundled static assets in `public/categories/`).
 * Keyed by the category slug. The 7 top-level categories are a fixed taxonomy,
 * so bundling beats a network/Cloudinary fetch (faster, works offline on
 * mobile, no per-icon round-trip on DRC 2G/3G). Subcategories — and any future
 * top category without an asset yet — fall back to the emoji.
 *
 * To add a category image: drop `public/categories/<slug>.png` and add one line.
 */
const CATEGORY_IMAGES: Record<string, string> = {
  supermarche: '/categories/supermarche.png',
  'telephones-et-accessoires': '/categories/telephones-et-accessoires.png',
  electromenager: '/categories/electromenager.png',
  mode: '/categories/mode.png',
  'beaute-et-sante': '/categories/beaute-et-sante.png',
  'construction-et-bricolage': '/categories/construction-et-bricolage.png',
  'automobile-et-moto': '/categories/automobile-et-moto.png',
};

export function categoryImage(slug?: string | null): string | null {
  if (!slug) return null;
  return CATEGORY_IMAGES[slug] ?? null;
}
